sessionctx/variable/session.go

```diff
@@ -1432,7 +1432,7 @@ func (s *SessionVars) SetSystemVar(name string, val string) error {
 	case TiDBDMLBatchSize:
 		s.DMLBatchSize = int(tidbOptInt64(val, DefOptCorrelationExpFactor))
 	case TiDBCurrentTS, TiDBLastTxnInfo, TiDBConfig:
-		return ErrReadOnly
+		return ErrReadOnly.GenWithStackByArgs(name, name, name)
```

Comment:
Is it possible to remove the `case`, and instead set the `ReadOnly: true` property of each of these sysvars? i.e. see the example in https://github.com/pingcap/tidb/issues/21943#issuecomment-749646780

By moving all sysVar validation to the struct, it makes it easier for plugins to be first class citizens.